Pinta/Actions/File/SaveDocumentImplementationAction.cs

```diff
@@ -189,6 +189,29 @@ private bool SaveFile (Document document, Gio.File? file, FormatDescriptor? form
 			return false;
 		}
 
+		// If the format doesn't support layers, ask to flatten the image
+		if (!format.SupportsLayers && document.Layers.Count() > 1)
+		{
+			string heading = Translations.GetString("This format does not support layers. Flatten image?");
+			string body = Translations.GetString("Flattening the image will merge all layers into a single layer.");
+
+			using var dialog = Adw.MessageDialog.New(chrome.MainWindow, heading, body);
+			dialog.AddResponse("cancel", Translations.GetString("_Cancel"));
+			dialog.AddResponse("flatten", Translations.GetString("Flatten"));
+			dialog.SetResponseAppearance("flatten", Adw.ResponseAppearance.Suggested);
+
+			dialog.CloseResponse = "cancel";
+			dialog.DefaultResponse = "flatten";
+
+			string response = dialog.RunBlocking();
+
+			if (response == "cancel")
+				return false;
+
+			// Flatten the image
+			PintaCore.Actions.Image.Flatten.Activate();
```

Comment:
I think the `tools.Commit ()` should probably happen before this block, in case a tool has uncommitted changes for one of the layers